crates/bevy_core_pipeline/src/core_2d/mod.rs

```diff
@@ -162,9 +245,61 @@ pub fn extract_core_2d_camera_phases(
 ) {
     for (entity, camera) in &cameras_2d {
         if camera.is_active {
-            commands
-                .get_or_spawn(entity)
-                .insert(SortedRenderPhase::<Transparent2d>::default());
+            commands.get_or_spawn(entity).insert((
+                SortedRenderPhase::<Opaque2d>::default(),
+                SortedRenderPhase::<Transparent2d>::default(),
+            ));
         }
     }
 }
+
+pub fn prepare_core_2d_depth_textures(
+    mut commands: Commands,
+    mut texture_cache: ResMut<TextureCache>,
+    msaa: Res<Msaa>,
+    render_device: Res<RenderDevice>,
+    views_2d: Query<
+        (Entity, &ExtractedCamera),
+        (
+            With<SortedRenderPhase<Opaque2d>>,
+            With<SortedRenderPhase<Transparent2d>>,
+            With<Camera2d>,
+        ),
+    >,
+) {
+    let mut textures = HashMap::default();
+    for (entity, camera) in &views_2d {
+        let Some(physical_target_size) = camera.physical_target_size else {
+            continue;
+        };
+
+        let cached_texture = textures
+            .entry(camera.target.clone())
+            .or_insert_with(|| {
+                // The size of the depth texture
+                let size = Extent3d {
+                    depth_or_array_layers: 1,
+                    width: physical_target_size.x,
+                    height: physical_target_size.y,
+                };
+
+                let descriptor = TextureDescriptor {
+                    label: Some("view_depth_texture"),
+                    size,
+                    mip_level_count: 1,
+                    sample_count: msaa.samples(),
+                    dimension: TextureDimension::D2,
+                    format: CORE_2D_DEPTH_FORMAT,
+                    usage: TextureUsages::RENDER_ATTACHMENT,
+                    view_formats: &[],
+                };
+
+                texture_cache.get(&render_device, descriptor)
+            })
+            .clone();
+
+        commands
+            .entity(entity)
+            .insert(ViewDepthTexture::new(cached_texture, Some(0.0)));
+    }
+}
```

Comment:
Going to give reviewing this PR a shot! My rendering knowledge is getting better, but I still have some gaps that I need to fill before signing off.

Currently my understanding of a depth buffer is more-or-less that of a texture, where the colors map to the "tallest" (highest) z-index of what was actually rendering at each pixel (vertex?).

But I'm not certain that understanding is correct, as this chunk of code is just grabbing... _something_ from the texture cache? That part is a bit of an unknown to me. I'm not sure where/how it actually gets written to.

Reply:
Your understanding sounds right to me.

This code just makes sure that the view will end up with the right type of texture attached as the depth buffer. Writing to it and using it happens on the GPU during rendering.

Reply:
Oh I see, it's a supported operation on the WebGPU side. We're just getting the texture ready 👍

Reply:
To add a bit more detail. The depth is generated by the output of the fragment shader. It will be automatically added to the texture when the fragment shader finishes. This code just sets up the texture for each 2d camera.

Reply:
Got it! Then we can easily use the `ViewDepthTexture` in a `ViewNode` 👍